pytest_fixtures/component/maintain.py

```diff
@@ -59,3 +58,35 @@ def module_synced_repos(sat_maintain):
     rh_repo.sync()
 
     yield {'custom': cust_repo, 'rh': rh_repo}
+
+
+@pytest.fixture
+def setup_sync_plan(request, sat_maintain):
+    """This fixture is used to create/delete sync-plan.
+    It is used by tests test_positive_sync_plan_disable_enable and test_positive_maintenance_mode.
+    """
+    org = sat_maintain.api.Organization().create()
+    # Setup sync-plan
+    new_sync_plan = sat_maintain.cli_factory.make_sync_plan(
+        {
+            'enabled': 'true',
+            'interval': 'weekly',
+            'organization-id': org.id,
+            'sync-date': datetime.datetime.today().strftime("%Y-%m-%d"),
+        }
+    )
+    sat_maintain.execute(f'cp {satellite_maintain_yml} foreman_maintain.yml')
+    sat_maintain.execute(f'sed -i "$ a :manage_crond: true" {satellite_maintain_yml}')
+
+    yield sat_maintain.api.SyncPlan(organization=org.label).search(query={'search': 'enabled=true'})
+
+    @request.addfinalizer
+    def _finalize():
+        assert sat_maintain.cli.MaintenanceMode.stop().status == 0
+        sat_maintain.execute(f'cp foreman_maintain.yml {satellite_maintain_yml}')
+        sat_maintain.execute(f'rm -rf {satellite_maintain_yml}')
```

Comment:
I think we want to remove the backup file here.
```suggestion
        sat_maintain.execute(f'rm -rf foreman_maintain.yml')
```

Reply:
Silly miss, updated, @vsedmik request to re-review